### ISSUE TITLE
Port Mapbase code

### ISSUE THREAD

**Issue by stepa2** [kallinosis-dev/srcmodbase-source#4]
Port Mapbase (from https://github.com/mapbase-source/source-sdk-2013).

- [ ] Merge Mapbase code with `git merge` (to be able to include further changes from Mapbase).
- [ ] Remove/merge backports from newer versions of Source Engine into Mapbase.
- [ ] Credit developers properly.

**Comment by ghost** [comment]
Whats the point of this?

**Comment by stepa2** [comment]
The point of this project is to make Source into a better modding platform/engine (better for me, at least, and probably it will be useful to others).
Mapbase has quite a lot of fixes and enchancements, so I'd like to use them in my project.
And I'd like to manage a merge to be able to do further merges on major Mapbase updates.

**Comment by ghost** [comment]
> The point of this project is to make Source into a better modding platform/engine (better for me, at least, and probably it will be useful to others). Mapbase has quite a lot of fixes and enchancements, so I'd like to use them in my project. And I'd like to manage a merge to be able to do further merges on major Mapbase updates.

Majority of mapbase is Source 2013, so git merging it will NOT be simple, nor work even if you do get it to compile. These are 2 different seperate engine branches

**Comment by stepa2** [comment]
I know, also this branch already has Valve-made VScript, entities that were backported into Mapbase and other conflicts.
But I think doing the merge probably will worth it.

**Comment by ghost** [comment]
> I know, also this branch already has Valve-made VScript, entities that were backported into Mapbase and other conflicts. But I think doing the merge probably will worth it.

DUDE, CSGO ALREADY HAD VSCRIPT FROM ALIEN SWARM, AND MAPBASES VSCRIPT IS ALIEN SWARM BUT BACKPORTED

**Comment by stepa2** [comment]
IIRC the implementation (not API) of VScript in Mapbase was written from scratch because is was not in the ASW SDK. But I may misremember.

> Mapbase supports a custom implementation of VScript based off of the Alien Swarm SDK and the public Squirrel API. It was originally created by reductor with the goal of creating a legal, open-source version of VScript in Source 2013. Mapbase's implementation of VScript does not use any leaked code whatsoever.

Also, the Valve-written bindings in game libs would probably conflict with Mapbase ones.

**Comment by ghost** [comment]
> IIRC the implementation (not API) of VScript in Mapbase was written from scratch because is was not in the ASW SDK. But I may misremember.
> 
> > Mapbase supports a custom implementation of VScript based off of the Alien Swarm SDK and the public Squirrel API. It was originally created by reductor with the goal of creating a legal, open-source version of VScript in Source 2013. Mapbase's implementation of VScript does not use any leaked code whatsoever.
> 
> Also, the Valve-written bindings in game libs would probably conflict with Mapbase ones.

I said it was backported from alien swarm dumbass

**Comment by stepa2** [comment]
You're wrong. 
The Mapbase docs say the implementation was written from scratch.
The ASW SDK lacks the implementation (see https://github.com/Source-SDK-Archives/source-sdk-alien-swarm).
Implementation and bindings in CSGO may differ from ASW. (The API seem to be same, at least the DLL interface version.)

And we're not on Dvach to call each other dumbasses.

**Comment by ghost** [comment]
Mapbase is based on Alien Swarms vscript, just heavily modified for Source 2013, same with CSGO since csgo is based on portal 2's engine branch which is based on ASW

**Comment by stepa2** [comment]
Docs and comments say different.
https://github.com/mapbase-source/source-sdk-2013/wiki/VScript-in-Mapbase
https://github.com/mapbase-source/source-sdk-2013/blob/master/sp/src/vscript/vscript.cpp#L1-L9

**Comment by ghost** [comment]
<img width="989" height="96" alt="Image" src="https://github.com/user-attachments/assets/bcde5f1f-8bc3-4b48-8133-6eb087403f63" />
Please for the love of god READ.

**Comment by stepa2** [comment]
Yes. "Based off" means it uses public API (the two header files in `public/vscript/`) and the bindings (not all of them, for example, if entity functions doesn't exist in Source 2013, bindings are probably not ported).

My point is that merging VScript in will still require some work (quite a lot of it, probably) due to divergent evolution of Mapbase and CSGO branch. And this whole merge is a later-time issue, because the engine doesn't fully compile now.

What is your point?
